client.js

```diff
@@ -45,3 +47,22 @@ document.addEventListener('DOMContentLoaded', function () {
     return new EditorToolbar(dom.find('[' + references.referenceAttribute + '*="/components/clay-kiln"]'));
   }
 });
+
+// handle connection loss
+
+window.addEventListener('load', function () {
+  // test connection loss on page load
+  if (!navigator.onLine) {
+    // we're offline!
+    progress.open('grey', connectionLostMessage);
+  }
+});
+
+window.addEventListener('online', function () {
+  progress.close(); // in case there are any status messages open, close them
+});
+
+window.addEventListener('offline', function () {
+  progress.done('grey'); // turn any progress indicators to grey and end them
+  progress.open('grey', connectionLostMessage);
+});
```

Comment:
What if the visual changes for this?  You're going to search for all reference to "grey"?

This event shouldn't reference the visual, but should represent what happened so people listening to the event can respond to a thing that has occurred.  That is, it's weird that people are responding to a "code grey".

"Alert!  Alert!  We've gone to Code Fuchsia!"  "What does that mean?"  "I don't know!!"